src/renderers/wgpu/node_bundle.rs

```diff
@@ -0,0 +1,166 @@
+use encase::ShaderType;
+use wgpu::{BindGroup, Device, RenderBundle};
+
+use crate::{
+    nodes::{
+        node::InoxNodeUuid,
+        node_data::{InoxData, Mask, Part},
+    },
+    puppet::Puppet,
+};
+
+use super::{
+    buffers::InoxBuffers,
+    pipeline::{CameraData, InoxPipeline, Uniform},
+};
+
+#[derive(Debug)]
+pub struct PartData(pub RenderBundle, pub Vec<Mask>);
+
+#[derive(Debug)]
+pub enum NodeBundle {
+    Part(PartData),
+    Composite(Vec<PartData>, InoxNodeUuid),
+}
+
+fn part_bundle_for_part(
+    device: &Device,
+    setup: &InoxPipeline,
+    buffers: &InoxBuffers,
+    model_texture_binds: &Vec<BindGroup>,
```

Comment:
Clippy gives a warning for this. It should be a slice
```suggestion
    model_texture_binds: &[BindGroup],
```